### PR TITLE
MCOL-5779: comment on alter table [stable 23.10]

### PR DESCRIPTION
**Pull request by mariadb-LeonidFedorov**
use encoding for correct check of ALTER TABLE CHANGE COLUMN

### CODE REVIEW

File: dbcon/ddlpackageproc/altertableprocessor.cpp [hunk 1]
```diff
@@ -88,6 +90,14 @@ struct extentInfo
 
 namespace
 {
+
+bool checkTextTypesLengthAreUqual(const CalpontSystemCatalog::ColType& colType, const ColumnType& newType)
```
Comment (drrtuy) [inline dbcon/ddlpackageproc/altertableprocessor.cpp]: Uqual - is this a typo?

File: dbcon/execplan/calpontsystemcatalog.h [hunk 2]
```diff
@@ -224,7 +224,7 @@ class CalpontSystemCatalog : public datatypes::SystemCatalog
     bool autoincrement = 0;  // set to true if  SYSCOLUMN autoincrement is �y�
     uint64_t nextvalue = 0;  // next autoincrement value
     uint32_t charsetNumber = default_charset_info->number;
-    const CHARSET_INFO* cs = nullptr;
+    const mutable CHARSET_INFO* cs = nullptr;
```
Comment (drrtuy) [inline dbcon/execplan/calpontsystemcatalog.h:227]: Are you sure we need a trick like this to make `getCharset` const? We actually changes the state of ColType instance

**Review by drrtuy — CHANGES_REQUESTED** [review]

File: dbcon/execplan/calpontsystemcatalog.h (hunk 2 above)
Comment (mariadb-LeonidFedorov) [reply dbcon/execplan/calpontsystemcatalog.h:227]: Yes, sure, getCharset initilizes this field if empty, I don't want to change the semantics

File: dbcon/ddlpackageproc/altertableprocessor.cpp (hunk 1 above)
Comment (mariadb-LeonidFedorov) [reply dbcon/ddlpackageproc/altertableprocessor.cpp]: This is Albanian accent, I'll fix

**Review by drrtuy — APPROVED** [review]
untested ACK

**Pull request merged**